pkg/resources/scim_integration.go

```diff
@@ -74,140 +86,170 @@ func SCIMIntegration() *schema.Resource {
 
-	// Some properties can come from the SHOW INTEGRATION call
-
-	s, err := snowflake.ScanScimIntegration(row)
+	integration, err := client.SecurityIntegrations.ShowByID(ctx, id)
 	if err != nil {
-		return fmt.Errorf("could not show security integration")
+		log.Printf("[DEBUG] notification integration (%s) not found", d.Id())
+		d.SetId("")
+		return diag.FromErr(err)
 	}
 
-	// Note: category must be Security or something is broken
-	if c := s.Category.String; c != "SECURITY" {
-		return fmt.Errorf("expected %v to be an Security integration, got %v", id, c)
+	if c := integration.Category; c != "SECURITY" {
+		return diag.FromErr(fmt.Errorf("expected %v to be a SECURITY integration, got %v", id, c))
 	}
 
-	if err := d.Set("scim_client", strings.TrimPrefix(s.IntegrationType.String, "SCIM - ")); err != nil {
-		return err
+	if err := d.Set("name", integration.Name); err != nil {
+		return diag.FromErr(err)
 	}
 
-	if err := d.Set("name", s.Name.String); err != nil {
-		return err
+	if err := d.Set("comment", integration.Comment); err != nil {
+		return diag.FromErr(err)
 	}
 
-	if err := d.Set("created_on", s.CreatedOn.String); err != nil {
-		return err
+	if err := d.Set("created_on", integration.CreatedOn.String()); err != nil {
+		return diag.FromErr(err)
 	}
 
-	// Some properties come from the DESCRIBE INTEGRATION call
-	// We need to grab them in a loop
-	var k, pType string
-	var v, unused interface{}
-	stmt = snowflake.NewSCIMIntegrationBuilder(id).Describe()
-	rows, err := db.Query(stmt)
-	if err != nil {
-		return fmt.Errorf("could not describe security integration")
+	if err := d.Set("enabled", integration.Enabled); err != nil {
+		return diag.FromErr(err)
 	}
-	defer rows.Close()
-	for rows.Next() {
-		if err := rows.Scan(&k, &pType, &v, &unused); err != nil {
-			return fmt.Errorf("unable to parse security integration rows")
+	typeParts := strings.Split(integration.IntegrationType, "-")
+	if len(typeParts) < 2 {
+		return diag.Diagnostics{
+			diag.Diagnostic{
+				Severity: diag.Error,
+				Summary:  "Invalid security integration type format.",
+				Detail:   fmt.Sprintf("expected \"SCIM - <scim_client>\", got: %s", integration.IntegrationType),
+			},
 		}
-		switch k {
+	}
+	scimClient := strings.TrimSpace(typeParts[1])
+	if err := d.Set("scim_client", scimClient); err != nil {
+		return diag.FromErr(err)
+	}
+	integrationProperties, err := client.SecurityIntegrations.Describe(ctx, id)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	for _, property := range integrationProperties {
+		name := property.Name
+		value := property.Value
+		switch name {
+		case "ENABLED", "COMMENT":
+			// We set this using the SHOW INTEGRATION call so let's ignore it here
 		case "NETWORK_POLICY":
-			if err := d.Set("network_policy", v.(string)); err != nil {
-				return fmt.Errorf("unable to set network policy for security integration")
+			if err := d.Set("network_policy", value); err != nil {
```

Comment:
network policy is an identifier, should we do some preprocessing before putting to state? (i.e. what if it is returned with quotes from SF?)

Reply:
But his means, it is put to state with double quotes. So if the user uses it without them in his config, this will cause a perma diff, right?

Reply:
Oh, okay, you have diff suppression for it. Let's leave it like this then. We will revisit it with identifiers rework